javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -360,6 +348,37 @@ public static ClientMethodMapper getInstance() {
         return methods;
     }
 
+    private void addClientMethodWithContext(List<ClientMethod> methods, Builder builder, ProxyMethod proxyMethod,
+        List<ClientMethodParameter> parameters, ClientMethodType clientMethodType, String proxyMethodName,
+        ReturnValue returnValue) {
+        ClientMethodParameter contextParam = new ClientMethodParameter.Builder()
+            .description("The context to associate with this operation.")
+            .wireType(ClassType.Context)
+            .name("context")
+            .annotations(new ArrayList<>())
+            .isConstant(false)
+            .defaultValue(null)
+            .fromClient(false)
+            .isFinal(false)
+            .isRequired(true)
+            .build();
+
+        List<ClientMethodParameter> withContextParameters = new ArrayList<>();
+        withContextParameters.addAll(parameters);
+        withContextParameters.add(contextParam);
+
+        methods.add(builder
+            .parameters(withContextParameters) // update builder parameters to include context
+            .returnValue(returnValue)
+            .name(proxyMethodName)
+            .onlyRequiredParameters(false)
+            .type(clientMethodType)
+            .isGroupedParameterRequired(false)
```

Comment:
Not sure whether need to set all these?

It seems to me that only thing need to be swapped is `withContextParameters`? All rest appears to be same as the `builder` from input parameter?

Reply:
Yes, you're right that the builder is already configured with these values from the previous client method but I didn't want to make an assumption that this method would be added only after the non-context method is added.